File: android/LedEqualizer/app/src/main/java/com/github/zeckson/ledequalizer/common/activities/SampleActivityBase.java
package com.github.zeckson.ledequalizer.common.activities;

import android.os.Bundle;
import android.support.v4.app.FragmentActivity;

import com.github.zeckson.ledequalizer.common.logger.Log;
import com.github.zeckson.ledequalizer.common.logger.LogWrapper;

public class SampleActivityBase extends FragmentActivity {

    public static final String TAG = "SampleActivityBase";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected  void onStart() {
        super.onStart();
        initializeLogging();
    }

    /** Set up targets to receive log data */
    public void initializeLogging() {
        LogWrapper logWrapper = new LogWrapper();
        Log.setLogNode(logWrapper);

        Log.i(TAG, "Ready");
    }
}

File: android/LedEqualizer/app/src/main/kotlin/com/github/zeckson/ledequalizer/common/activities/SampleActivityBase.kt
package com.github.zeckson.ledequalizer.common.activities

import android.os.Bundle
import android.support.v4.app.FragmentActivity

import com.github.zeckson.ledequalizer.common.logger.Log
import com.github.zeckson.ledequalizer.common.logger.LogWrapper

open class SampleActivityBase : FragmentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onStart() {
        super.onStart()
        initializeLogging()
    }

    /** Set up targets to receive log data  */
    open fun initializeLogging() {
        val logWrapper = LogWrapper()
        Log.logNode = logWrapper

        Log.i(TAG, "Ready")
    }

    companion object {

        val TAG = "SampleActivityBase"
    }
}
